config/lobby/db/001_create_tables

```diff
@@ -1,94 +1,58 @@
-create table bad_words (
-    word character varying(40) not null
+CREATE TABLE ta_users (
+    username character varying(40) NOT NULL PRIMARY KEY,
+    password character(34),
+    bcrypt_password character(60),
+    CHECK (password IS NOT NULL OR bcrypt_password IS NOT NULL),
+    email character varying(254) NOT NULL,
+    joined timestamp with time zone DEFAULT CURRENT_TIMESTAMP,
+    lastlogin timestamp with time zone DEFAULT CURRENT_TIMESTAMP,
+    admin integer NOT NULL DEFAULT 0
 );
+ALTER TABLE ta_users owner TO postgres;
 
-
-alter table bad_words owner to postgres;
-
-
-create table banned_ips (
-    ip character varying(40) not null,
-    ban_till timestamp without time zone
+CREATE TABLE bad_words (
+    word character varying(40) NOT NULL PRIMARY KEY
 );
+ALTER TABLE bad_words owner TO postgres;
 
 
-alter table banned_ips owner to postgres;
-
-
-create table banned_macs (
-    mac character varying(40) not null,
-    ban_till timestamp without time zone
+CREATE TABLE banned_ips (
+    ip character varying(40) NOT NULL PRIMARY KEY,
+    ban_till timestamp with time zone
 );
+ALTER TABLE banned_ips owner TO postgres;
 
 
-alter table banned_macs owner to postgres;
-
-
-create table banned_usernames (
-    username character varying(40) not null,
-    ban_till timestamp without time zone
+CREATE TABLE banned_macs (
+    mac character varying(40) NOT NULL PRIMARY KEY,
+    ban_till timestamp with time zone
 );
+ALTER TABLE banned_macs owner TO postgres;
 
 
-alter table banned_usernames owner to postgres;
-
-create table muted_ips (
-    ip character varying(40) not null,
-    mute_till timestamp without time zone
+CREATE TABLE banned_usernames (
+    username character varying(40) NOT NULL PRIMARY KEY REFERENCES ta_users ON DELETE CASCADE ON UPDATE CASCADE,
```

Comment:
I added a reference, so in case a username is changed/removed while it's banned somehow, the entry is being updated/removed along with it.
I don't understand why a seperate table was even created for this.
banned_usernames and ta_users have a 1:1 relation, which means they should be merged permanently.

Also banned_* and muted_* could be merged to a single table with 3 columns each

Reply:
the username field in ta_users is presumably mutable, there is no relationship between the two at all. If we ban a name, then a user deletes and recreates their account, we want to keep the username banned. Seems like this would delete the entry with ta_users and so a banned username could be defeated by recreating an account.


I think in general we should avoid cascade deletes/updates. Fundamentally I think the real problem is that there is no id column for users.

Reply:
I thought about the ID problem, but I didn't want to increase the scope of this PR that much.
I'd suggest  assigning a pseudo-random id instead of a simple increasing one, in case we want to create an API some time in the future.
```
CREATE SEQUENCE sequence_name;
CREATE TABLE ta_users (
    id bigint NOT NULL DEFAULT pseudo_encrypt(nextval('sequence_name')),
    ...
);
```
> If we ban a name, then a user deletes and recreates their account, we want to keep the username banned. Seems like this would delete the entry with ta_users and so a banned username could be defeated by recreating an account.

AFAIK there is no way to delete your own account and even if there was a way, how would we know that the new account isn't someone who did randomly pick the same name as someone else?
The updating cascade however seems pretty useful to me. If a user renames his account, the ban is instantly applied to the new username instead.

Reply:
I suggest punting on this, we really don't need to solve this exact problem now, it's not related to passwords, and there is a direction we want to go with DB to support game challenges and some player tracking. Right before that, dropping parts and/or simplifying the schema would be a good stepping stone. 

In the meantime this quite a lot to take in with the context of bcrypt hashing. 

> The updating cascade however seems pretty useful to me. If a user renames his account, the ban is instantly applied to the new username instead.

That is not the intent of the feature, that really changes things. The idea is so that offensive user names would be blocked, it's not meant to track specific user accounts. It's trivial to create new user accounts.